fix(ios): memory leak in require (8_0_X)

janvennemann opened the pull request:
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-26783

8_0_X backport of #10659

janvennemann committed:
fix(ios): memory leak in require

vijaysingh-axway commented on iphone/TitaniumKit/TitaniumKit/Sources/API/KrollBridge.m:
```diff
@@ -1366,7 +1366,7 @@ - (id)require:(KrollContext *)kroll path:(NSString *)path
   NSURL *oldURL = [self currentURL];
   NSString *workingPath = [oldURL relativePath];
-  ResolvedModule *resolved = [self resolveRequire:path withWorkingPath:workingPath];
+  ResolvedModule *resolved = [[self resolveRequire:path withWorkingPath:workingPath] autorelease];
```
Isn't it good if we use autorelease, where we are allocating the object. So better we should go inside function`-(ResolvedModule *)resolveRequire:(NSString *)path withWorkingPath:(NSString *)workingPath ` and use autorelease while allocating.

build commented:
<!--
  1 failure:  :microscope: Ther...
  0 warning: 
  2 messages
  
  DangerID: danger-id-default;
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  :microscope: There are library changes, but no changes to the unit tests. That's OK as long as you're refactoring existing code, but will require an admin to merge this PR. Please see <a href="https://github.com/janvennemann/titanium_mobile/blob/TIMOB-26783-8_0_X/README.md#unit-tests">README.md#unit-tests</a> for docs on unit testing.
  </td>
    </tr>
  </tbody>
</table>



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  :floppy_disk: <a href="https://jenkins.appcelerator.org/job/titanium-sdk/job/titanium_mobile/job/PR-10660/1/artifact/dist/mobilesdk-8.0.0.v20190130101823-osx.zip">Here's the generated SDK zipfile</a>.
  </td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>

  :white_check_mark: All tests are passing
Nice one! All 2990 tests are passing.
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ba5f30cd1a13a5dc31f456e4f3523d86bd3eafb9
</p>

janvennemann committed:
send autorelease on alloc

This also shuts up the static analyzer since the methods are not adhering to the naming convention for methods that return +1 references.

vijaysingh-axway reviewed (approved):
CR passed.

sgtcoolguy commented:
This PR should be merged post 8.0.0.GA since the fix on master has already been merged.

sgtcoolguy committed:
Merge branch '8_0_X' into TIMOB-26783-8_0_X

sgtcoolguy committed:
Merge branch '8_0_X' into TIMOB-26783-8_0_X
